test/jdk/java/foreign/TestMemoryInspection.java

```diff
@@ -0,0 +1,288 @@
+
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @enablePreview
+ * @modules java.base/jdk.internal.foreign
+ * @run testng/othervm --enable-native-access=ALL-UNNAMED TestMemoryInspection
+ */
+
+import java.lang.foreign.*;
+import java.lang.invoke.VarHandle;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import jdk.internal.foreign.MemoryInspection;
+import org.testng.annotations.*;
+
+import static java.lang.foreign.ValueLayout.*;
+import static java.util.stream.Collectors.joining;
+import static org.testng.Assert.*;
+import static java.util.Objects.requireNonNull;
+
+@Test
+public class TestMemoryInspection {
```

Comment:
The test is good, but it never initialized the segments to something meaningful. As such, the test will not detect issues e.g. when parsing elements of an array (e.g. the array test uses an array of 4 structs, but since the contents of the segment is all zeros, all the structs have same values).